src/components/NavigationHeader.tsx

```diff
@@ -20,36 +16,31 @@ function NavigationHeader({
   const title = t(getHeaderTitle(options, route.name));
 
   return (
-    <Flex height={layout.height / 10 + HEADER_HEIGHT} justifyContent="flex-end">
-      <Pressable
-        style={{ flex: 1 }}
+    <Flex
+      flexDirection="row"
+      justifyContent="space-between"
+      alignItems="center"
+      backgroundColor="palette.neutral.c00"
+      py={6}
+      mb={6}
+    >
+      {back ? (
+        <Link size="large" Icon={ArrowLeftMedium} onPress={navigation.goBack} />
+      ) : (
+        <View />
+      )}
+      <Text variant="large" fontWeight="semiBold">
+        {title}
+      </Text>
+      <Link
+        size="large"
+        Icon={CloseMedium}
         onPress={() => {
-          navigation.canGoBack() && navigation.goBack();
+          navigation.getParent()?.goBack();
         }}
       />
-      <Flex
-        flexDirection="row"
-        justifyContent="space-between"
-        alignItems="center"
-        backgroundColor="palette.neutral.c00"
-        p={6}
-      >
-        {back ? (
-          <Link
-            size="large"
-            Icon={ArrowLeftMedium}
-            onPress={navigation.goBack}
-          />
-        ) : (
-          <View />
-        )}
-        <Text variant="large" fontWeight="semiBold">
-          {title}
-        </Text>
-        <Link size="large" Icon={CloseMedium} onPress={navigation.popToTop} />
-      </Flex>
     </Flex>
   );
 }
 
-export default NavigationHeader;
+export default (props: StackHeaderProps) => <NavigationHeader {...props} />;
```

Comment:
Why not just return the component? 🤔

Reply:
Because it causes hook issues due to a poor react navigation implementation. :/